doc/developer/guide-changes.md

```diff
@@ -0,0 +1,699 @@
+according to Git's merge algorithm, because they don't touch the same lines. But
+when both changes are present, the code won't compile, because the new call site
+uses the old signature for the function. The analogous situation arises when
+one PR adds a new test while another PR adjusts the behavior tested by that
+new test.
+
+Fixing merge skew is usually as simple as identifying the above situation
+and adjusting the new function call or test case to account for the change
+in the other PR.
+
+While there _are_ tools that can prevent merge skew, like [Bors], they introduce
+quite a bit of latency and flakiness when merging a PR. As long as merge skew is
+fairly rare—right now it seems to happen once every month or two—net
+productivity is still higher without Bors.
+
+#### Branch restrictions
+
+In very rare cases, it may be necessary to merge a PR whose tests are failing.
+This situation typically only occurs with chicken-and-egg CI configuration
+issues, where PR builds are broken until a fix can be merged to `main`.
+
+To perform a "god mode" merge, you can temporarily lift the branch restrictions
+on the `main` branch. Navigate to the [branch protection rules for the `main`
+branch][branch-protection] and uncheck the "Include administrators" box. Do
+*not* delete the rules or uncheck any other boxes, as it is much harder to
+reverse those changes.
+
+Your PR will be mergeable after you uncheck the "Include administrators" box,
+though you will have to click through several warning screens. Once the PR is
+merged, be sure to re-enable the "Include administrators" setting to prevent
+accidents.
+
+## Reviewing changes
+
+Google's overriding principle for code reviewers is as follows:
+
+> Reviewers should favor approving a PR once it definitely improves the overall
+> health of the codebase, even if it is not perfect.
+
+This principle does a good job balancing the need of engineers to be
+productive with the need of code owners to keep the codebase maintainable.
+
+### Ensuring codebase health
+
+See the dedicated [Software engineering
+principles](#software-engineering-principles) section for the general software
+engineering principle we strive to enforce. When applying our software
+engineering principles indicates that there is clear room for improvement on a
+PR, leave a comment to that effect!
+
+Remember that codebase health often degrades over time as a result of a number
+of small changes, each of which introduce a small amount of complexity. The
+amount of complexity can seem justified when viewed in isolation, but over time,
+the complexity can accrete to result in very confusing, unmaintainable code.
+
+Some matters can be more subjective. When you have a preference, but reasonable
+folks can disagree, prefix your comment with "nit:" to indicate that it is not
+a hard requirement for the PR to merge.
+
+### Mentoring
+
+When someone is new to Rust, code review is one of the primary components of
```

Comment:
[potentially out of scope]: everything in this section talks about how the code reviewer helps the reviewee who is new to Rust / new to this part of the codebase. But imo doing code reviews is also very beneficial to the code reviewer as well. 

In my experience, reviewing the work of engineers more and less senior than me helps me understand how to better think about code, reviewing changes in a part of the codebase I am unfamiliar with helps me ramp up on that part, and doing code reviews helps me if in the future there are bugs / customer issues with that part of the code.

I suspect that none of this is controversial, but I think it might be nice to include in this doc somewhere (doesn't have to be here) to help motivate those readers who might be less inclined to do code reviews (especially if we hire less experienced people in the future)?

Reply:
Good idea! Added to the "Choosing reviewers" section up above.